### ISSUE TITLE
Error 500 when deleting too many records

### ISSUE THREAD

**Issue by rodrigoscoelho** [espocrm/espocrm#998]
As you can see on the animated gif below, EspoCRM is with big problems when dealing with too many records at the same time. 

![erro espocrm](https://user-images.githubusercontent.com/6912410/43661638-63240fa4-9739-11e8-9c17-d5a1f55b1a3e.gif)

**Comment by bandtank** [comment]
What happens in the error log? Did any of the records actually delete or did it break before the database was touched?

**Comment by yurikuzn** [comment]
Hi,

Definitely there must be something in log.

**Comment by bandtank** [comment]
You may also want to check the webserver's logs. You are going to have to post some kind of error statement.

**Comment by yurikuzn** [comment]
I believe it's a memory limit issue.

**Comment by bandtank** [comment]
That's what I was wondering as well. This huge of a deletion may need to be done via sql queries directly or via workflow with loops and smaller data sets.

**Comment by yurikuzn** [comment]
With sql queries directly is not an option. We need to check an access to each record before delete.

**Comment by rodrigoscoelho** [comment]
[Fri Aug 03 16:29:14.569578 2018] [proxy_fcgi:error] [pid 26599:tid
140361701627648] [client 200.233.228.218:7304] AH01071: Got error 'PHP
message: PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted
(tried to allocate 4096 bytes) in
/var/www/html/espocrm/application/Espo/Services/Record.php on line 1444\n',




2018-08-03 16:30 GMT-03:00 Yuri Kuznetsov <notifications@github.com>:

**Comment by bandtank** [comment]
Sure, if you need the ACL direct sql won't work. I mentioned it just in case that wasn't important. I sometimes delete things out of the database using pdo queries because I don't care about the ACL.

I didn't mean as a general purpose delete method. I meant for a specific requirement.

**Comment by yurikuzn** [comment]
BTW, old jobs are deleted with a cleanup job.

**Comment by yurikuzn** [comment]
Actually, it shouldn't take such a lot of memory. I will look into.

**Comment by rodrigoscoelho** [comment]
Not cleaning here.

Take a look at the size of that database:
![image](https://user-images.githubusercontent.com/6912410/43662096-1994b35a-973b-11e8-8c7e-697929a70dbd.png)

**Comment by yurikuzn** [comment]
It's a normal size. Find the oldest one. It must be not older than 2 months back.

**Comment by bandtank** [comment]
It cleans on a schedule. How many records are being created per day in your job table? It's possible that it is being cleaned, but new records are being created very frequently.

**Comment by rodrigoscoelho** [comment]
you are right. The oldest one is from June 23

**Comment by rodrigoscoelho** [comment]
Yes, I check emails each minute. That's why we have a lot of job records

**Comment by rodrigoscoelho** [comment]
each 30 seconds, actually.

**Comment by bandtank** [comment]
How are you checking emails that frequently?

**Comment by yurikuzn** [comment]
You can set in config 
`'cleanupJobPeriod ' => '15 days'`

**Comment by yurikuzn** [comment]
Job can't be run more frequently than once per minute. We need a daemon to make it run more often.

**Comment by rodrigoscoelho** [comment]
1 minute, sorry. 2 records each time because we have 2 personal accounts.
